### PR TITLE
Fix: InceptionBN.__call__

### PR DESCRIPTION
**Pull request by cympfh**
BatchNormalization (http://docs.chainer.org/en/stable/_modules/chainer/links/normalization/batch_normalization.html#BatchNormalization) can set test-mode on `__call__(self, x, test=False)`,
but InceptionBN (http://docs.chainer.org/en/stable/_modules/chainer/links/connection/inceptionbn.html#InceptionBN) has a difference interface `__call__(self, x)` and stores test-mode as a property.
The difference confuses us.

### CODE REVIEW

**Comment by delta2323** [comment]
Thank you for suggestion. I agree with you in that identical APIs for `BatchNormalization` and `InceptionBN` reduce the confusion. On the other hand, It is difficult for us to remove `train` property because we have the [policy for the API compatibility](http://docs.chainer.org/en/stable/compatibility.html). One of the solution is to offer both ways to set the train/test mode of `InceptionBN` and define the priority between the two (e.g. if the `test` argument of `__call__` is explicitly set, we adopt it).

By the way, we make use of the fact that `train` mode is set through the attribute in the ImageNet example ([here](https://github.com/pfnet/chainer/blob/master/examples/imagenet/train_imagenet.py#L141) is the code). If we remove the `train` property or make it obsolete, it is better not to use it.

**Comment by cympfh** [comment]
I tried to keep the compatibility.

Following 2 codes are same.

```
model.__call__(x, test=bl)
```

```
model.train = not bl
model.__call__(x)
```

**Comment by delta2323** [comment]
Thanks! I will check your change.

**Comment by delta2323** [comment]
We discussed within the developer team and concluded that it is confusing for users to be provided two ways to do the same thing (in this case, to set the test mode through the attribute and through the argument of `__call__` method). So, it is better to choose the one as the recommended way and make the other deprecated.

On the other hand, it is much confusing to change the "recommended way" within short terms, so we should discuss this point enough before making a change. I would like your opinion on this problem. Thanks in advance.

**Comment by delta2323** [comment]
@beam2d Could you check this PR?

**Comment by beam2d** [comment]
I'll check it shortly.

**Comment by beam2d** [comment]
Note that I'm preparing a training mode configuration in #2186 and then this change will not be needed in v2, and so this change can be left unmerged for v2. It is still meaningful to merge it to v1. @delta2323 How do you think of it?

**Comment by delta2323** [comment]
I agree that we do not need this feature any longer in v2.

I come think it is OK to merge in v1 if we add sufficient explanations. I concerned the multiple standard ways of setting the training mode could confuse users. But I come to think it is acceptable if we write explicitly that there are two ways to do the same thing and that it is for keeping backward compatibility.

**Comment by delta2323** [comment]
I decided to merge this PR to v1. We will send another PR for v2 that renews its implementation.
